### ISSUE TITLE
JSX parsing errors out on valid HTML structure

### ISSUE THREAD

**Issue by DanCouper** [solidjs/solid#1491]
The error it's throwing indicates it's due to [one of the] Babel [plugin dependencies], but the solid plugin is the instigator. So this:

```
<dl>
  <div>
    <dt>Thing</dt>
    <dd>Description of thing</dd>
  </div>
</dl>
```

Errors like

```
[plugin:solid] /Users/daniel.couper/Work/PERSONAL/pscan/src/app.tsx: Invalid JSX: <dt> cannot be child of <div>
  76 |         <dl class="appheader__meta">
  77 |           <div class="appheader__metaitem">
> 78 |             <dt class="appheader__metaitem-k">Version</dt>
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  79 |             <dd class="appheader__metaitem-v">{version}</dd>
  80 |           </div>
  81 |           <div class="appheader__metaitem">
```

Which is annoying, because it's been spec for 5 years. I'm trying to drill through to where exactly it's blowing up, but I'm not terribly familiar with how vite plugins & then the babel plugins they depend on glue together. Stacktrace:

```
    at File.buildCodeFrameError (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/core/lib/transformation/file/file.js:205:12)
    at NodePath.buildCodeFrameError (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/index.js:105:21)
    at JSXElement (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/babel-plugin-jsx-dom-expressions/index.js:3241:20)
    at NodePath._call (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:45:20)
    at NodePath.call (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:35:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:80:31)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitMultiple (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:61:17)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:107:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:86:52)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitMultiple (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:61:17)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:107:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:86:52)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitMultiple (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:61:17)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:107:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:86:52)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitMultiple (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:61:17)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:107:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:86:52)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitSingle (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:65:19)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:109:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:86:52)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitMultiple (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:61:17)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:107:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:86:52)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitSingle (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:65:19)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:109:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:86:52)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitSingle (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:65:19)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:109:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17)
    at NodePath.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/path/context.js:86:52)
    at TraversalContext.visitQueue (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:86:16)
    at TraversalContext.visitMultiple (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:61:17)
    at TraversalContext.visit (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/context.js:107:19)
    at traverseNode (/Users/daniel.couper/Work/PERSONAL/pscan/node_modules/@babel/traverse/lib/traverse-node.js:18:17
```

**Comment by DanCouper** [comment]
> This discussion is probably more applicable in the [`validate-html-nesting`](https://github.com/MananTank/validate-html-nesting) repository

Yep (and thanks @ryansolid for raising the issue). The updating of the transformer is possibly triggered things?? Last time I tried to do this was in React a year or so ago, and the Babel transformer was involved there and there were no issues (IIRC! Memory is slightly hazy, testing library was the thing I was having issues with). 

I don't think this is correct though:

> the HTML validator says you can't put a `div` inside of a `dl` because it's a "[Flow content](https://html.spec.whatwg.org/multipage/dom.html#flow-content)" element. [The HTML validator is throwing errors](https://validator.w3.org/nu/#textarea) when I try to do anything but nest `dt` and `dl` as direct children. More on the spec can be found [here](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) which DOES cite the ability to provide a wrapping `div`... but... the [validator](https://validator.w3.org/nu/#textarea) has conflicting opinions.

What are you entering to make it error? This validates:

```
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>
  <dl>
    <div>
      <dt>Hi</dt>
      <dd>Something about hi</dd>
    </div>

    <div>
      <dt>Ho</dt>
      <dd>Something about ho</dd>
    </div>

    <div>
      <dt>Hiho</dt>
      <dd>Something about hi</dd>
      <dd>Something about ho</dd>
    </div>
  </dl>
</body>
</html>
```

**Comment by ryansolid** [comment]
Ok this should be fixed in version `1.2.1` which is in the range so package lock clear and clean install should get you the latest.

**Comment by russelgal** [comment]
how do I disable the validator for certain files?
